protocol-definitions/ReplicatedMap.yaml

```diff
@@ -820,3 +820,113 @@ methods:
             since: 2.0
             doc: |
               Number of entries affected by this event.
+  - id: 19
+    name: putAllWithMetadata
+    since: 2.7
+    doc: |
+      Copies all of the mappings from the specified entry list to this map, including metadata.
+      This method uses ReplicatedRecordStore.putRecords method with version=current version + 1.
+      Please note that all the keys in the request should belong to the partition id to which this request is being sent, all keys
+      matching to a different partition id shall be ignored. The API implementation using this request may need to send multiple
+      of these request messages for filling a request for a key set if the keys belong to different partitions.
+    request:
+      retryable: false
+      partitionIdentifier: any key belongs to target partition
+      params:
+        - name: name
+          type: String
+          nullable: false
+          since: 2.7
+          doc: |
+            name of replicated map
+        - name: entries
+          type: List_ReplicatedMapEntryViewHolder
+          nullable: false
+          since: 2.7
+          doc: |
+            entries with metadata
+        - name: version
+          type: long
+          nullable: false
+          since: 2.7
+          doc: |
+            version of the replicated map store that stores the data
+    response: {}
+  - id: 20
+    name: fetchEntryViews
+    since: 2.7
+    doc: |
+      Allows iteration over ReplicatedMapEntryViewHolder objects. The changes happened during the iteration may not be included in the iterated EntryViews. This method
+      will throw an exception if there is no replicated record store with the given replicated map name and partition id. This method will consume some memory in the 
```

Comment:
how much is this `some memory`? If I have many concurrent requests how is memory usage growing?

Reply:
Depends on the entry sizes and fetchSize parameter.